ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -1612,9 +1613,31 @@ func (r *RayClusterReconciler) updateRayClusterStatus(ctx context.Context, origi
 	if err != nil {
 		logger.Info("Error updating status", "name", originalRayClusterInstance.Name, "error", err, "RayCluster", newInstance)
 	}
+	collectRayClusterMetrics(r.options.RayClusterMetricCollector, newInstance.Name, newInstance.Namespace, newInstance.Status, originalRayClusterInstance.Status, originalRayClusterInstance.CreationTimestamp.Time)
+
 	return inconsistent, err
 }
 
+// collectRayClusterMetrics records metrics related to the RayCluster.
+func collectRayClusterMetrics(collector *metrics.RayClusterMetricCollector, name, namespace string, newClusterStatus, oldClusterStatus rayv1.RayClusterStatus, creationTimestamp time.Time) {
+	if collector == nil {
+		return
+	}
+
+	// Record `kuberay_cluster_provisioned_duration_seconds` metric if just provisioned
+	if meta.IsStatusConditionTrue(newClusterStatus.Conditions, string(rayv1.RayClusterProvisioned)) &&
+		!meta.IsStatusConditionTrue(oldClusterStatus.Conditions, string(rayv1.RayClusterProvisioned)) {
+		collector.ObserveRayClusterProvisionedDuration(name, namespace, time.Since(creationTimestamp).Seconds())
+	}
+}
+
+// DeleteRayClusterMetrics deletes all metrics related to the RayCluster.
+func DeleteRayClusterMetrics(collector *metrics.RayClusterMetricCollector, name, namespace string) {
```

Comment:
maybe we don't need to export this function.
```suggestion
func deleteRayClusterMetrics(collector *metrics.RayClusterMetricCollector, name, namespace string) {
```